src/map/ai/controllers/trust_controller.cpp

```diff
@@ -71,6 +73,11 @@ void CTrustController::Tick(time_point tick)
 
 void CTrustController::DoCombatTick(time_point tick)
 {
+    if (POwner->PMaster == nullptr || POwner->PMaster->isDead())
```

Comment:
Can you test replacing the movement stuff in `DoCombatTick` with `CMobController::Move()`?  Since this is now derived from CMobController (which it always should have been), there's a bunch of stuff that may be unnecessarily duplicated in this file